centerofci/mathesar
# Frontend - update all of our npm packages

This PR updates almost all of our npm packages. I've verified and fixed all package update related breaking changes. 

**Before reviewing**
* Stop the server
* Pull in branch
* Delete your local node_modules folder and the node_modules shared docker volume
* Start the server
* If you run into any problem, purge your docker container and build again

**Checklist of things to verify for reviewers**
* Run dev server. Check if page loads and test general sanity on any of the pages.
* Build with `npm run build` and run Mathesar in production mode. Check if page loads and test general sanity.
* These should be covered already by our CI:
   * `npm run typecheck`
   * `npm run test`
   * `npm run check-format`
   * `npm run lint`
* Run `npm run storybook` and `npm run build-storybook` and test general sanity on storybook.

## Checklist
 
  index.md`).
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
